lib/cequel/metal/keyspace.rb

```diff
@@ -316,7 +316,12 @@ def extract_hosts_and_port(configuration)
       end
 
       def extract_credentials(configuration)
-        configuration.slice(:username, :password).presence
+        credentials = configuration.slice(:username, :password).presence
+        
+        return credentials unless configuration[:auth_provider]
+
+        { auth_provider: CequelPassword.new(*credentials.values),
+          datacenter: configuration[:datacenter].presence }
```

Comment:
That seems pretty reasonable. I went down this route because I didn't want to force the behavior, unsure if always including a provider would break an expectation. If its not a concern that I think its the right way to go.